adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/_ui/responsive/common/js/adyen.checkout.js

```diff
@@ -329,17 +336,64 @@ var AdyenCheckoutHybris = (function () {
             });
 
             try {
-                paypalComponent.mount(paypalNode);
+                this.adyenComponent.mount(paypalNode);
             } catch (e) {
                 console.log('Something went wrong trying to mount the PayPal component: ' + e);
             }
         },
 
-        makePayment: function (form, component, handleResult) {
+        initiateMbway: function () {
+            var mbwayNode = document.getElementById('adyen-component-container');
+
+            this.adyenComponent = this.checkout.create("mbway", {
+                showPayButton: false,
+                onChange: (state, component) => {
+                    if (!state.isValid) {
```

Comment:
If user provides valid details, place order will be enabled. what if user changes the details and it is not valid? we are not explicitly first disabling placeorder onchange action.

Reply:
The flow here is:
- When the page is first loaded, the component is rendered and the "place order" button is enabled 
- When the button is clicked, it's disabled to prevent multiple payment submissions (check "configureButton" fuction). This is specially needed for MB Way, since the user remains on the same page while the component shows the payment progress.
- When the button is clicked, the "onChange" action on the component is called. Here we are doing the "isValid" check, and if the form is not valid (i.e. invalid/not provided email or telephone number), the "Place Order" button is enabled again so the payment can be submitted again after the user fixed the form data.
- If everything isValid, the flow continues on the "onSubmit" action and the payment is submitted